apps/meteor/ee/server/lib/oauth/Manager.ts

```diff
@@ -50,9 +50,12 @@ export class OAuthEEManager {
 				user.roles = [];
 			}
 
+			const savedRoles = Roles.findInIdsOrNames(rolesToSync).toArray();
+			const allowedRoles = savedRoles.map((role) => role._id);
```

Comment:
```suggestion
			const savedRoles = Promise.await(Roles.findInIdsOrNames(rolesToSync));
			const allowedRoles = Promise.await(savedRoles.toArray()).map((role) => role._id);
```
I think we're just missing some awaits here